sessions/2018/november/Session_17.md

```diff
@@ -0,0 +1,79 @@
+# SOFTWARE ENGINEERING IN JAVA
+
+## Session 17 (13/11/2018)
+
+- Java API backend and JavaScript+HTML+CSS frontend application demo.
+
+### Notes
+
+Roger was showing me an example of an application he developed in the past, this was the game called Hangman, that consists in guess a word with a maximum of tries.
+
+The code of the Java API Backend is on <https://github.com/kinbiko/hangman> and the Frontend code is on <https://github.com/kinbiko/hangman-frontend>
+
+#### Backend
+
+##### Groovy
+
+The backend application was written with a service in [Groovy][11] which is a language based in Java but with several advantages such as simpler variable definition, not need of final semicolons and several useful methods such as `bytes` as is shown following:
+
+```groovy
+final String stringRepresentation = new String(new File("src/main/resources/passwords.txt").bytes)
+```
+
+In this case that piece of code is extracting the content of the `passwords.txt` as bytes and and placing it in a `String ` variable.
+
+Groovy is a good language to learn but is dependant on be updated with Java versions, because Java is the dominant programming language.
+
+Other languaged based o Java are Kotlin and Scala.
+
+
+##### JavaScript vs CofeeScript
+
+Another example of a Programming Languaje based in another is how CofeeScript and TypeScript whic are based on javaScript ([some details here][4]). About [CofeeScript][10], a real life example was gave by Roger in how the things can complicate with CofeeScript. When CofeeScript born at the end of 2009, it was intended to be a solution to all JavaScript problems at that time, but also introduced syntax changes. With the pass of the time, JavaScript solved the problems had and improved but in a different way CofeeScript did, so this made separate paths between both of them plus all the syntax differences. Some of the arguments against CofeeScript can be found [here][5] and [here][6].
+
+##### Domain Specific Language   
+
+Domain Specific Languages (DSL) are small programming languages used inside an application with the purpose to execute certain focused tasks into applications, examples of them are SQL, CSS, Regular Expressions. Gradle scripts written with Groovy are other exampke if a DLS, that means an DLS can be written in another programming language that isn't a DSL.
```

Comment:
s/exampke/example/

So seems like you've got a good understanding of this. Just to add some terminology around the last point in this line: When a DSL is based on another language (like gradle) it's called an 'internal DSL'. Conversely, when it is a language completely separate from any other languages it's called an 'external DSL'.